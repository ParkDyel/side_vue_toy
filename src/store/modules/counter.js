import {SET_COUNTER, GET_COUNTER} from "@/store/mutation-types"

export const counter = {
  namespace:true,
  state :{
    counter:0
  },
  actions:{
    [SET_COUNTER] : ({commit}) => commit(SET_COUNTER)
  },
  mutations:{
    [SET_COUNTER] : state => state.counter++
  },
  getters:{
    [GET_COUNTER]:state => state.counter
  }
}
