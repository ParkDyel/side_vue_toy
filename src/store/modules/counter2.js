import {SET_COUNTER, GET_COUNTER2} from "@/store/mutation-types"

export const counter2 = {
  namespace:true,
  state :{
    counter2:0
  },
  actions:{
    [SET_COUNTER] : ({commit}) => commit(SET_COUNTER)
  },
  mutations:{
    [SET_COUNTER] : state => state.counter2++
  },
  getters:{
    [GET_COUNTER2]:state => state.counter
  }
}
