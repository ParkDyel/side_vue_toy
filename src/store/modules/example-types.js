// store/modules/
export const GET_COUNTER = 'GET_COUNTER';
export const SET_COUNTER = 'SET_COUNTER';
