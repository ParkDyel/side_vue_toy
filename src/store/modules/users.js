import axios from 'axios';
import { REGISTER_USER } from './types';
export const userStore = {
  namespaced: true,
  state: {},
  getters: {},
  actions: {
    [REGISTER_USER]: async ({ commit, data }) => {
      let res = await axios.post('/api/v1/user', { data });
    },
  },
  mutations: {},
};
