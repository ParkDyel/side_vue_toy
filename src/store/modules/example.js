// store/modules/example
import { GET_COUNTER, SET_COUNTER } from './example-types';
export const exampleStore = {
  state: {
    counter: 0,
  },
  getters: {
    [GET_COUNTER]: (state) => state.counter,
  },
  mutations: {
    [SET_COUNTER]: (state) => state.counter++,
  },
  actions: {
    [SET_COUNTER]: ({ commit }) => {
      commit(SET_COUNTER);
    },
  },
};
