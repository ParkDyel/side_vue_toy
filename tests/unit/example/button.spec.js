// tests/unit/example/button.spec.js
import { shallowMount, createLocalVue } from '@vue/test-utils';
import ButtonPage from '@/components/example/ButtonPage';
import Vuex from 'vuex';
import { exampleStore } from '@/store/modules/example';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('ButtonClick', () => {
  let wrapper;
  let mockSomeMethod;
  let store;
  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        exampleStore,
      },
    });
    mockSomeMethod = jest.spyOn(ButtonPage.methods, 'someMethod');
    wrapper = shallowMount(ButtonPage, { store, localVue });
  });
  it('Page 스냅샷을 찍자', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('기본 영역을 확인하자 : 숫자 나오는 부분', () => {
    const count = wrapper.find('.count-area').text();
    expect(count).toBe('0');
  });

  it('기본 영역을 확인하자 : 버튼 렌더 부분', () => {
    const btn = wrapper.find('.increment-btn');
    expect(btn.text()).toBe('Click Me');
  });

  it('함수의 로직은 정상 작동 하는가', () => {
    let res = wrapper.vm.doSomething(2);
    expect(res).toBe(true);
  });

  it('버튼의 함수가 제대로 호출 되는가', () => {
    const btn = wrapper.find('.increment-btn');
    btn.trigger('click');
    expect(mockSomeMethod).toHaveBeenCalled();
  });

  it('버튼의 함수가 인자와 함께 호출 되는가', () => {
    const btn = wrapper.find('.increment-btn');
    btn.trigger('click');
    expect(mockSomeMethod).toBeCalledWith('name');
  });

  it('버튼을 클릭하면 counter가 +1 증가되어야한다', () => {
    const btn = wrapper.find('.increment-btn');
    btn.trigger('click');
    expect(store.state.exampleStore.counter).toEqual(1);
  });
});
