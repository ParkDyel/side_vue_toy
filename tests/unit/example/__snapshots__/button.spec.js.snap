// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonClick Page 스냅샷을 찍자 1`] = `
<div>
  <div class="count-show"><span class="count-area">0</span></div>
  <div><button class="increment-btn">Click Me</button></div>
</div>
`;
